maven-api-impl/src/main/java/org/apache/maven/internal/impl/model/DefaultModelBuilder.java

```diff
@@ -595,6 +595,8 @@ Model transformFileToRaw(Model model) {
                     } else {
                         newDeps.add(dep);
                     }
+                } else {
+                    newDeps.add(dep);
                 }
             }
             return modified ? model.withDependencies(newDeps) : model;
```

Comment:
I feel that remembering `modified` no longer makes sense and obfuscates what this method is doing. What purpose of that variable am I missing?

Reply:
The point is to not recreate a new `Model` object is there's no need to.  We could go further and delay the creation of the list until it's actually needed, but that would mean store the index, etc... I think the code was less readable.